docs/corruption-detection-and-recovery.md

```diff
@@ -0,0 +1,62 @@
+# etcd Corruption Detection and Recovery
+
+Currently, in etcd 3.5.0-3.5.2, there is a low probability risk of data corruption across the etcd members [Upstream Issue](https://github.com/etcd-io/etcd/issues/13766), [Openshift Documentation](https://access.redhat.com/solutions/6849521). There is a mitigation coming to the Openshift Cluster etcd Operator that enables the etcd 'experimental-initial-corrupt-check' which should help detect corruption and stop the bad member from joining the cluster.
+This document will outline some ways to verify if an etcd member is corrupted so that you can be more confident that the corruption recovery steps are necessary and should be taken.
+
+# Corruption Diagnosis
+Unfortunately, there's no single command that will give a precise 'corrupt/not corrupt' at this time; however, if most or all of the following return that there might be corruption, then it's likely that there is corruption and recovery steps should be taken. Please note, that a single failure in the following commands does not necessarily mean that corruption has occurred.
+
+## Check Prometheus for etcd revision divergence
+Revision divergence is a natural happening of etcd since it takes time to sync a revision to all members; however, continuing divergence is an indication of corruption. If you have access to run Prometheus queries, you can check the following to graph the current revisions to visualize any divergence.
+
+`max by (pod,job) (etcd_debugging_mvcc_current_revision)`
+
+---
+
+## Get dbSize and revision differences
+One indicator that there might be corruption is a 'significant' (greater than 10%) difference of etcd database size and revision number.
```

Comment:
I don't see how you'd get into this state without accumulating significant revision divergence.  It seems very unlikely to me that two members with such wildly divergent storage sizes happen to coincidentally agree on the revision number for very long.  But doesn't hurt to check this one too; maybe I'm wrong and you can get this without significant divergence.  I wouldn't want folks to delay their diagnosis just based on "the size hasn't gotten different enough yet", since that would allow a good deal of split-brain divergence to accumulate.